lms/djangoapps/lms_xblock/test/test_settings_service.py

```diff
@@ -0,0 +1,76 @@
+    """ 'Static' class to hold constants used in `override_settings` decorators in TestSettingsService """
+    xblock_setting_key1 = 'dummy_block'
+    xblock_setting_key2 = 'other_dummy_block'
+
+
+class _DummyBlock(object):
+    """ Dummy Xblock class """
+    pass
+
+
+@ddt.ddt
+class TestSettingsService(TestCase):
+    """ Test SettingsService """
+    def setUp(self):
+        """ Setting up tests """
+        super(TestSettingsService, self).setUp()
+        self.settings_service = SettingsService()
+        self.xblock_mock = mock.Mock()
+        self.xblock_mock.block_settings_key = _Constants.xblock_setting_key1
+        self.xblock_mock.unmixed_class = mock.Mock()
+        self.xblock_mock.unmixed_class.__name__ = _Constants.xblock_setting_key2
+
+    def test_get_given_none_throws_value_error(self):
+        """  Test that given None throws value error """
+        with self.assertRaises(ValueError):
+            self.settings_service.get_settings_bucket(None)
+
+    def test_get_return_none_or_default_if_xblock_settings_is_missing(self):
+        """ Test that returns default (or None if default not set) if XBLOCK_SETTINGS is not set """
+        self.assertFalse(hasattr(settings, 'XBLOCK_SETTINGS'))  # precondition check
+        self.assertIsNone(self.settings_service.get_settings_bucket(self.xblock_mock))
+        self.assertEqual(self.settings_service.get_settings_bucket(self.xblock_mock, 'zzz'), 'zzz')
+
+    @override_settings(XBLOCK_SETTINGS={_Constants.xblock_setting_key2: {'b': 1}})
+    def test_get_returns_none_or_default_if_bucket_not_found(self):
+        """ Test if settings service returns default if setting not found """
+        self.assertEqual(getattr(settings, 'XBLOCK_SETTINGS'), {_Constants.xblock_setting_key2: {'b': 1}})
+        self.assertIsNone(self.settings_service.get_settings_bucket(self.xblock_mock))
+        self.assertEqual(self.settings_service.get_settings_bucket(self.xblock_mock, 123), 123)
+
+    @override_settings(XBLOCK_SETTINGS={_Constants.xblock_setting_key1: {'a': 1}})
+    def test_get_returns_correct_value(self):
+        """ Test if settings service returns correct bucket """
+        self.assertEqual(getattr(settings, 'XBLOCK_SETTINGS'), {_Constants.xblock_setting_key1: {'a': 1}})
+        self.assertEqual(self.settings_service.get_settings_bucket(self.xblock_mock), {'a': 1})
+
+    @override_settings(XBLOCK_SETTINGS={_Constants.xblock_setting_key2: "I'm a setting"})
+    def test_get_respects_block_settings_key(self):
+        """ Test if settings service respects block_settings_key value """
+        self.assertEqual(getattr(settings, 'XBLOCK_SETTINGS'), {_Constants.xblock_setting_key2: "I'm a setting"})
+        self.xblock_mock.block_settings_key = _Constants.xblock_setting_key2
+        self.assertEqual(self.settings_service.get_settings_bucket(self.xblock_mock), "I'm a setting")
+
+    @override_settings(XBLOCK_SETTINGS={_DummyBlock.__name__: "I'm another setting"})
```

Comment:
Nit: I think this would be better if you just used `"_DummyBlock"` as a string instead of `_DummyBlock.__name__` in this test. Doing so would be more clear, more distinct from the code we're testing, and is how these settings will actually be entered into `settings.py` files (people will be entering them as strings, not as `MyXBlock.__name__`).